bundles/org.openhab.core.persistence/src/test/java/org/openhab/core/persistence/extensions/PersistenceExtensionsTest.java

```diff
@@ -42,53 +52,72 @@
  */
 public class PersistenceExtensionsTest {
 
-    public static final String TEST_NUMBER = "Test Number";
-    public static final String TEST_SWITCH = "Test Switch";
-
-    @NonNullByDefault
-    private final PersistenceServiceRegistry registry = new PersistenceServiceRegistry() {
-
-        private final PersistenceService testPersistenceService = new TestPersistenceService();
-
-        @Override
-        public @Nullable String getDefaultId() {
-            // not available
-            return null;
-        }
-
-        @Override
-        public @Nullable PersistenceService getDefault() {
-            // not available
-            return null;
-        }
+    public static final String TEST_NUMBER = "testNumber";
+    public static final String TEST_QUANTITY_NUMBER = "testQuantityNumber";
+    public static final String TEST_SWITCH = "testSwitch";
 
-        @Override
-        public Set<PersistenceService> getAll() {
-            return Set.of(testPersistenceService);
-        }
+    private @Mock UnitProvider unitProvider;
+    private @Mock ItemRegistry mockedItemRegistry;
 
-        @Override
-        public @Nullable PersistenceService get(@Nullable String serviceId) {
-            return TestPersistenceService.ID.equals(serviceId) ? testPersistenceService : null;
-        }
-    };
+    private AutoCloseable mocksCloseable;
 
-    private CoreItemFactory itemFactory;
-    private GenericItem numberItem, switchItem;
+    private GenericItem numberItem, quantityItem, switchItem;
 
     @BeforeEach
     public void setUp() {
-        new PersistenceExtensions(registry);
-        itemFactory = new CoreItemFactory();
+        mocksCloseable = openMocks(this);
```

Comment:
When the class is annotated with `@ExtendWith(MockitoExtension.class)`, the `mockCloseable` and `afterEach` method can be removed.

Reply:
Deon.

Once upon a time when I submitted my changes it was not possible at all ... 🤣

Reply:
There was still an issue with using it in integration tests which is fixed by https://github.com/openhab/openhab-core/pull/2075 . For some complicated setups of tests extending another class it may still make sense to use it.

Reply:
Yes, I am aware of that. Was just kidding.